File: chain-of-responsibility/java/BaseHandler.java
package ChainOfResponsibility;

// Zincirin bir sonraki halkasının tanımlandığı sınıftır.
// İşlemi gerçekleştirecek olan metot tanımı bulunur.
// UML diyagramındaki BaseHandler sınıfına denk gelmektedir.
abstract class BaseHandler {
    protected BaseHandler nextHandler;

    public void setNextHandler(BaseHandler nextHandler) {
        this.nextHandler = nextHandler;
    }

    public abstract void handleRequest(Image image);
}

File: chain-of-responsibility/typescript/base-handler.ts
import {Image} from "./image";

// Zincirin bir sonraki halkasının tanımlandığı sınıftır.
// İşlemi gerçekleştirecek olan metot tanımı bulunur.
// UML diyagramındaki BaseHandler sınıfına denk gelmektedir.
export abstract class BaseHandler {
  protected nextHandler: BaseHandler;

  public setNextHandler(nextHandler: BaseHandler): void {
    this.nextHandler = nextHandler;
  }

  public abstract handleRequest(image: Image): void;
}
